src/libcore/alloc/layout.rs

```diff
@@ -258,19 +258,49 @@ impl Layout {
 
     /// Creates a layout describing the record for `self` followed by
     /// `next`, including any necessary padding to ensure that `next`
-    /// will be properly aligned. Note that the resulting layout will
-    /// satisfy the alignment properties of both `self` and `next`.
+    /// will be properly aligned, but *no trailing padding*. In order
+    /// to match C representation layout `repr(C)`, you should call
```

Comment:
Can you put the guidance about matching `repr(C)` in a separate paragraph?

Reply:
Done.